spring-cloud-stream-schema-server/src/main/java/org/springframework/cloud/stream/schema/server/controllers/ServerController.java

```diff
@@ -126,6 +133,49 @@ public ServerController(SchemaRepository repository,
 		return new ResponseEntity<>(schema, HttpStatus.OK);
 	}
 
+	@RequestMapping(value = "/{subject}/{format}/v{version}", method = RequestMethod.DELETE)
+	public void delete(@PathVariable("subject") String subject,
+			@PathVariable("format") String format,
+			@PathVariable("version") Integer version) {
+		if (this.schemaServerProperties.isAlllowSchemaDeletion()) {
+			Schema schema = this.repository.findOneBySubjectAndFormatAndVersion(subject, format,
+					version);
+			deleteSchema(schema);
+		}
+		else {
+			throw new SchemaDeletionNotAllowedException();
+		}
+	}
+
+	@RequestMapping(value = "/schemas/{id}", method = RequestMethod.DELETE)
+	public void delete(@PathVariable("id") Integer id) {
+		if (this.schemaServerProperties.isAlllowSchemaDeletion()) {
+			Schema schema = this.repository.findOne(id);
+			deleteSchema(schema);
+		}
+		else {
+			throw new SchemaDeletionNotAllowedException();
+		}
+	}
+
+	@RequestMapping(value = "/{subject}", method = RequestMethod.DELETE)
+	public void delete(@PathVariable("subject") String subject) {
+		if (this.schemaServerProperties.isAlllowSchemaDeletion()) {
```

Comment:
I am wondering though whether we shouldn't return a 405 response instead of just throwing an exception.